### ISSUE TITLE
Missadventure when upgrading to crosssigning

### ISSUE THREAD

**Issue by deepbluev7** [vector-im/riot-web#13188]
### Description

This is a pretty general description of what I experienced, when upgrading my old account with multiple devices to crosssigning, since I was asked by @ara4n to write it down somewhere.

First I upgraded my self hosted version of riot web to 1.6.0-rc1. I had crosssigning still set to labs in my riot config, but I had enabled it in my riot already. I never did the upgrade though.

Then I reopened riot-web in my Firefox on my Linux Laptop. I was prompted to upgrade my account or I clicked upgrade in the settings (not sure anymore). After that, I had 13 sessions to verify:

- The first one was my Nheko session on my Laptop. I didn't want to verify it later, so I hit verify. After that I only had 2 buttons to chose from: "Not my device" and "Verify". The later one only led to an infinite spinner, since Nheko doesn't support that verification yet. There should probably be an abort or skip button also, for clients that don't support verifying via emojis yet. Anyway, I figured out, that I could close the dialog, without breaking anything in any obvious way.

- After hitting later a few times, I came to a session, where I'm pretty sure it was a session of one of the computers around me. I hit verify, but I just got an infinite spinner. This happened to all my other devices. I couldn't verify any session from my Laptop.

- To debug the verification requests seemingly not arriving, I added "crosssigning: enabled" to my riot config and refreshed my riot tabs on my other devices. I was then prompted on both devices to verify my other session. Only my Windows laptop verification request actually arrived at my laptop though. The request from Firefox on my Linux desktop never arrived at my Laptop. I managed to verify my session on my desktop by logging into my secret store though.

- I also couldn't verify my MiniVector and RiotAndroid sessions. The requests never arrived. (I verified them manually before the upgrade though, so maybe that is the issue?)

- Someone then gave me the hint, that I could do a manual text verification from my own profile page. I would have never guessed that. I was looking through the settings and couldn't find a way to verify my sessions manually, since the prompts also lack that option.

- After that I tried to verify a User, that didn't upgrade to cross signing. That somewhat worked. It verified one of his devices, but it told me, it would verify his identity, which is mildly confusing.

Once I worked around those issues, crosssigning is pretty awesome though!

### Issues summary

- The toasts in the top left block access to the settings.
- Too many toasts make the whole flow a bit confusing especially since most don't have a back button, like the verification prompts in the upper left.
- There is no global "Verify all sessions later" button from what I can tell.
- No way to do legacy verification when upgrading.
- Verification requests (to devices verified before the upgrade maybe?) don't arrive (in almost all cases)
- Not having crosssigning in the settings doesn't seem to default it to on.
- There is no hint, that you are verifying a device instead of a user, when the other party doesn't do crosssigning yet.

### Version information

- **Platform**: web

For the web app:

- **Browser**: Firefox 75
- **OS**: Linux, Linux, Windows
- **URL**: riot.neko.dev, 1.6.0-rc1
- **homeserver**: neko.dev, Synapse 1.12.3

**Comment by jryans** [comment]
Thanks for the detailed feedback! Although you did hit a number of rough spots, I don't see anything to block release here, so we'll dig through this in more detail post-release.

I believe one change we'll make that should help with bits of this is ensuring cross-signing is actually enabled without requiring a config change.

**Comment by deepbluev7** [comment]
Yeah, probably not a blocker. The config change would be great (especially for self hosted riots). Apart from that, all the issues I had are solveable, they are just usability issues for new or less tech savvy users. But that can be fixed with the usability improvements, that are planned for after the cross-signing release anyway. :D

**Comment by deepbluev7** [comment]
Closing this as I can't provide feedback on this anymore and so many things changed, that this is probably just additional clutter.
